packages/core/src/CML/coreToCml/coreToCml.ts

```diff
@@ -436,17 +448,39 @@ export const txWitnessBootstrap = (
   return witnesses;
 };
 
-export const witnessSet = (scope: ManagedFreeableScope, witness: Cardano.Witness): TransactionWitnessSet => {
-  const cslWitnessSet = scope.manage(TransactionWitnessSet.new());
-  const vkeyWitnesses = scope.manage(Vkeywitnesses.new());
+const redeemerTagMap = {
+  [RedeemerPurpose.spend]: RedeemerTag.new_spend(),
+  [RedeemerPurpose.mint]: RedeemerTag.new_mint(),
+  [RedeemerPurpose.certificate]: RedeemerTag.new_cert(),
+  [RedeemerPurpose.withdrawal]: RedeemerTag.new_reward()
+};
 
-  if (witness.scripts) {
-    const { nativeScripts, plutusV1Scripts, plutusV2Scripts } = getScripts(scope, witness.scripts);
+export const txWitnessRedeemers = (scope: ManagedFreeableScope, redeemers: Cardano.Redeemer[]): Redeemers => {
+  const witnessRedeemers = scope.manage(Redeemers.new());
+  for (const redeemer of redeemers) {
+    const tag = redeemerTagMap[redeemer.purpose];
+    const index = BigNum.from_str(redeemer.index.toString());
+    const data = PlutusData.from_bytes(Buffer.from(redeemer.data, 'hex'));
+    const units = ExUnits.new(
+      BigNum.from_str(redeemer.executionUnits.memory.toString()),
+      BigNum.from_str(redeemer.executionUnits.steps.toString())
+    );
+    witnessRedeemers.add(Redeemer.new(tag, index, data, units));
```

Comment:
Might be a leak here
```suggestion
    witnessRedeemers.add(scope.manage(Redeemer.new(
      tag: scope.manage(tag.clone()),
      index: scope.manage(index),
      data: scope.manage(data),
      units: scope.manage(units)
    )));
```